sources/web/datalab/static/page.js

```diff
@@ -0,0 +1,231 @@
+function placeHolder() {}
+
+function reportEvent(event) {
+  var reportingEnabled = (document.body.getAttribute('data-reporting-enabled') == 'true');
+  if (!reportingEnabled) { return; }
+
+  var signedIn = (document.body.getAttribute('data-signed-in') == 'true');
+  var additionalMetadata = 'signedIn=' + signedIn;
+  if (event['metadata']) {
+    event['metadata'] = event['metadata'] + ',' + additionalMetadata;
+  } else {
+    event['metadata'] = additionalMetadata;
+  }
+  window.dataLayer = window.dataLayer || [];
+  window.dataLayer.push(event);
+}
+
+function toggleSidebar() {
```

Comment:
Sounds good. I'm actually surprised this seems to be the only place we got things tangled up, given we've been editing this massive file for a long time now.